packages/stable/src/api/instances/instancesApi.ts

```diff
@@ -93,4 +101,208 @@ export class InstancesAPI extends BaseResourceAPI<NodeOrEdge> {
       );
     return response.data;
   };
+
+  /**
+   * [Retrieve instances](https://developer.cognite.com/api#tag/Instances/operation/byExternalIdsInstances)
+   *
+   * ```js
+   *  const view = {
+   *    externalId: 'Describable',
+   *    space: 'cdf_core',
+   *    type: 'view',
+   *    version: 'v1',
+   *  };
+   *  const response = await client.instances.retrieve({
+   *    sources: [{ source: view }],
+   *     items: [
+   *       {
+   *         externalId: describable1.externalId,
+   *         space: testSpace.space,
+   *         instanceType: 'node',
+   *       },
+   *     ],
+   *   });
+   * ```
+   */
+  public retrieve = async (
+    params: ListOfSpaceExternalIdsRequestWithTyping
+  ): Promise<NodeAndEdgeCollectionResponseV3Response> => {
+    const response = await this.post<NodeAndEdgeCollectionResponseV3Response>(
+      this.byIdsUrl,
+      {
+        data: params,
+      }
+    );
+    return response.data;
+  };
+
+  /**
+   * [Upsert instances](https://developer.cognite.com/api#tag/Instances/operation/applyNodeAndEdges)
+   *
+   * ```js
+   *  const view = {
+   *    externalId: 'Describable',
+   *    space: 'cdf_core',
+   *    type: 'view',
+   *    version: 'v1',
+   *  };
+   *  await client.instances.upsert({
+   *   items: describables.map((describable) => ({
+   *       instanceType: 'node',
+   *       externalId: describable.externalId,
+   *       space: describable.space,
+   *       sources: [
+   *         {
+   *           source: view,
+   *           properties: {
+   *             title: describable.title,
+   *             description: describable.description,
+   *             labels: describable.labels,
+   *           },
+   *         },
+   *       ],
+   *     })),
+   *   });
+   * ```
+   */
+  public upsert = async (
+    params: NodeAndEdgeCreateCollection
+  ): Promise<SlimNodeAndEdgeCollectionResponse> => {
+    const response = await this.post<SlimNodeAndEdgeCollectionResponse>(
+      this.url(),
+      {
+        data: params,
+      }
+    );
+    return response.data;
+  };
+
+  /**
+   * [Delete instances](https://developer.cognite.com/api#tag/Instances/operation/deleteBulk)
+   *
+   * ```js
+   *  await client.instances.delete({
+   *   items: [
+   *      {
+   *        instanceType: 'node',
+   *        externalId: describable1.externalId,
+   *        space: testSpace.space,
+   *      },
+   *      {
+   *        instanceType: 'node',
+   *        externalId: describable2.externalId,
+   *        space: testSpace.space,
+   *      },
+   *    ],
+   *   })
+   * ```
+   */
+  public delete = async (params: ListOfSpaceExternalIdsRequestWithTyping) => {
+    await this.delete(params);
+  };
```

Comment:
FYI, in the other SDK methods for upserting, deleting, etc., we support deleting more instances than the hard API limit. We do this by chunking the array into `limit`-sizes and, in parallel, deleting each chunk.
I noticed this when working on the FDM SDK that we don't do this.
E.g. here is the delete method for assets supporting deleting >1000 assets in a single SDK call: https://github.com/cognitedata/cognite-sdk-js/blob/master/packages/stable/src/api/assets/assetsApi.ts#L121-L127